File: android/mobilepasssdk/src/main/java/com/armongate/mobilepasssdk/model/LocationRequirement.java
package com.armongate.mobilepasssdk.model;

/**
 * Location requirements for verification
 * Provided when remote access requires location check
 */
public class LocationRequirement {
    private final double latitude;
    private final double longitude;
    private final int radius;

    public LocationRequirement(double latitude, double longitude, int radius) {
        this.latitude = latitude;
        this.longitude = longitude;
        this.radius = radius;
    }

    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public int getRadius() {
        return radius;
    }
}

File: ios/MobilePassSDK/Model/LocationRequirement.swift
//
//  LocationRequirement.swift
//  MobilePassSDK
//
//  Created by Mobile Pass SDK
//

import Foundation

public struct LocationRequirement {
    public var latitude: Double
    public var longitude: Double
    public var radius: Int  // meters
    
    init(latitude: Double, longitude: Double, radius: Int) {
        self.latitude = latitude
        self.longitude = longitude
        self.radius = radius
    }
}

